src/main/java/seedu/address/storage/JsonAdaptedEvent.java

```diff
@@ -0,0 +1,141 @@
+
+    /**
+     * Constructs a {@code JsonAdaptedTask} with the given person details.
+     */
+    @JsonCreator
+    public JsonAdaptedEvent(@JsonProperty("name") String name, @JsonProperty("startDate") String startDate,
+                           @JsonProperty("startTime") String startTime,
+                           @JsonProperty("endDate") String endDate, @JsonProperty("endTime") String endTime,
+                           @JsonProperty("category") List<JsonAdaptedCategory> category,
+                           @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
+        this.name = name;
+        this.startDate = startDate;
+        this.startTime = startTime;
+        this.endDate = endDate;
+        this.endTime = endTime;
+        if (category != null) {
+            this.category.addAll(category);
+        }
+        if (tagged != null) {
+            this.tagged.addAll(tagged);
+        }
+    }
+
+    /**
+     * Converts a given {@code Task} into this class for Jackson use.
+     */
+    public JsonAdaptedEvent(Event source) {
+        name = source.getName().toString();
+        startDate = source.getStartDate().toString();
+        startTime = source.getStartTime().toString();
+        endDate = source.getEndDate().toString();
+        endTime = source.getEndTime().toString();
+        category.addAll(source.getCategories().stream()
+                .map(JsonAdaptedCategory::new)
+                .collect(Collectors.toList()));
+        tagged.addAll(source.getTags().stream()
+                .map(JsonAdaptedTag::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted person object into the model's {@code Event} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted person.
+     */
+    public Event toModelType() throws IllegalValueException {
+        final List<Category> eventCategories = new ArrayList<>();
+        final List<Tag> eventTags = new ArrayList<>();
+
+        for (JsonAdaptedTag tag : tagged) {
+            eventTags.add(tag.toModelType());
+        }
+
+        for (JsonAdaptedCategory cat : category) {
+            eventCategories.add(cat.toModelType());
+        }
+
+        if (name == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName()));
+        }
+        if (!Name.isValidName(name)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final Name modelName = new Name(name);
+
+        if (startDate == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Date.class.getSimpleName()));
+        }
+        if (!Date.isValidDate(startDate)) {
+            throw new IllegalValueException(Date.MESSAGE_CONSTRAINTS);
+        }
+        final Date modelStartDate = new Date(startDate);
+
+        if (startTime == null) {
```

Comment:
Time is not a compulsory field for event right? Only date is if Im not wrong! Is it necessary to throw missing field message format?